.github/workflows/container-image.yml

```diff
@@ -264,6 +261,15 @@ jobs:
         uses: actions/download-artifact@37930b1c2abaa49bbe596cd826c3c89aef350131
         with:
           name: "action_image_artifact_samba-server_${{ env.IMG_TAG }}"
+      - name: Set up k3d
+        run: |
+          set -x
+          curl --silent --fail -o install_k3d.sh https://raw.githubusercontent.com/rancher/k3d/main/install.sh
+          TAG="" bash install_k3d.sh
+          k3d --version
+          k3d cluster create --wait
+          while ! kubectl get serviceaccount default >/dev/null; do sleep 1; done
+          kubectl get nodes
```

Comment:
IMHO, it would be better to have this entire logic in a dedicated script (shell or python) with proper logging upon failure.